docs/02-hugo.Rmd

```diff
@@ -1,28 +1,83 @@
 
 Although we think Hugo is a fantastic static site generator, there is really one and only one important but missing feature: the support for R Markdown. That is basically the whole point of the **blogdown** package. This missing feature means that you cannot easily generate results using R code on your web pages, since you can only use static Markdown documents. Besides, you have to use Hugo's choice of the Markdown engine named "Blackfriday" instead of the more powerful Pandoc.^[There is a feature request in Hugo's repository that has been three years old: https://github.com/spf13/hugo/issues/234, and it seems that it will not be implemented in the near future.]
 
-What does a typical file and directory structure look like?
 
-- config.toml
-- content
-- static
-- themes
-- layouts
+Hugo uses a consistent file and folder structure to create your website.  The rest of this chapter will give more detail on the following files and folders:
+
+- `config.toml`
+- `content/`
+- `static/`
+- `themes/`
+- `layouts/`
+
+```{r folders, fig.cap='Files and Folders created when you create a new site using **blogdown**', echo=FALSE, fig.align='center', out.width='90%'}
+knitr::include_graphics('images/folder-structure.png')
+```
 
 ## Configuration
 
-TOML syntax
+The first file that Hugo uses to generate a website is the configuration or `config` file. It is worth mentioning here that technically speaking, Hugo could generate your website without a `config` file if your files and folders are in the correct locations. Regardless, Hugo will look for a `config` file in the root directory of your website, so it is best to make sure that you have one. Generally speaking, this is the only file that Hugo expects to be in the root directory. 
+
+As its name suggests, the `config` file is where you tell Hugo exactly how to render your webpage and display your menus. It is also the location of other global parameters like links to your social networks, your site's name, and the base URL for your website. 
+
+When generating your site, Hugo will search for a file called `config.toml` first. If it doesn't find one, it will continue to search for either `config.yaml` or `config.json`. Since most themes automatically generate a `config.toml` file, we will discuss that here. 
```

Comment:
At this point, I'd like to make a comment about the Hugo docs before introducing the TOML syntax: I think one very confusing thing about the Hugo documentation is that it is utterly inconsistent in syntax when giving examples. Sometimes it shows YAML, and sometimes it shows TOML (worse, a few examples are even in JSON). This can be frustrating to readers. I think we need to remind readers of this (when seeing `foo = 1`, it is TOML; when `foo: 1`, it is YAML; forget about JSON).

I don't know what you think, but TOML actually makes no sense to me. The advantages of TOML compared to YAML are totally trivial. I'm no big fan of YAML, either, but if I have to choose one, I will definitely choose YAML, since most R Markdown users are already familiar with it. What I'd recommend is that blogdown users use TOML only for config.toml, and that is only because the vast majority of themes provide `config.toml` in their example sites. For .md and .Rmd pages/posts, always use YAML for metadata. In fact, Rmd does not support TOML at all.

FWIW, my own website uses YAML even for the config file: https://github.com/rbind/yihui.name/blob/master/config.yaml